### DISCUSSION TITLE
[dev-cleanup] Refactor perf energy model

### DISCUSSION THREAD

**Discussion by deyuan** [UVA-LavaLab/PIMeval-PIMbench#143] [PIMeval]
Split class per PIM architecture

**Comment by fasiddique** [comment]
I am wondering should we create two base classes for performance and energy. Each PIM architecture will extend the base classes as required.

**Reply by deyuan** [reply]
There are lots of common code between performance and energy formulas, so I'm leaning to bundling perf and energy together for now. Another consideration is that we can first break perf energy models into a few major families, which can be extended or inherited later. 

I've created a PR #156 to dev-cleanup branch for early review. Feel free to share any thoughts or ideas.

**Comment by deyuan** [comment]
Closed with PR #156, #164, #167
